### PR TITLE
chore: Remove `@aws-cdk/aws-lambda-python-alpha`

### PR DESCRIPTION
**Pull request by xonstone**
### What does this PR do?

This PR simplifies the definition of the v2 LambdaFunction. This is possible because all specialised CDK Lambda functions (like NodeJSFunction, PythonFunction, ... ) extend from the generic CDK Lambda Function. 

### Motivation

The peer dependency on @aws-cdk/aws-lambda-python-alpha is causing a lot of issues since this is a pinned dependency due to it being published as an alpha. This will resolve tickets like the following:

- https://github.com/DataDog/datadog-cdk-constructs/issues/91
- https://github.com/DataDog/datadog-cdk-constructs/issues/118
- https://github.com/DataDog/datadog-cdk-constructs/issues/145
- https://github.com/DataDog/datadog-cdk-constructs/issues/208
- https://github.com/DataDog/datadog-cdk-constructs/issues/214

### Testing Guidelines

Ran all unit-test

Deployed the sample v2 stack and verified if the layers are still being added correctly

### Types of Changes

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] Misc (docs, refactoring, dependency upgrade, etc.)

### Check all that apply

- [x] This PR's description is comprehensive
- [ ] This PR contains breaking changes that are documented in the description
- [ ] This PR introduces new APIs or parameters that are documented and unlikely to change in the foreseeable future
- [ ] This PR impacts documentation, and it has been updated (or a ticket has been logged)
- [x] This PR's changes are covered by the automated tests
- [ ] This PR collects user input/sensitive content into Datadog

### CODE REVIEW

File: v2/package.json [hunk 1]
```diff
@@ -136,4 +135,4 @@
     "@types/babel__traverse": "7.18.2"
   },
   "//": "~~ Generated by projen. To modify, edit .projenrc.js and run \"npx projen\"."
-}
+}
```
Comment (duncanista) [inline v2/package.json:138]: nit
```suggestion
}

```

File: v2/package.json [hunk 2]
```diff
@@ -75,7 +74,7 @@
   },
   "main": "lib/index.js",
   "license": "Apache-2.0",
-  "version": "0.0.0",
+  "version": "1.9.0",
```
Comment (duncanista) [inline v2/package.json:77]: autogenerated on release
```suggestion
  "version": "0.0.0",
```

**Review by duncanista — COMMENTED** [review]
Thanks for your contribution – left some minor comments.

**Comment by xonstone** [comment]
Thanks for the review! Fixed the open comments.

**Comment by xonstone** [comment]
@duncanista Any chance you can take another look?

**Comment by duncanista** [comment]
Hey @xonstone, I'm ready to approve, yet, I'm afraid I won't be able to release until next week due to an active Code Freeze on our side. If this is business critical for you, please raise a ticket with us, and make sure to add this PR and this comment. Normally, we wouldn't request this, but since I'm currently limited on the release side, I have to ask for this if this is business critical for you. 

Yet, rest assured that we are continuing with this and following up on it!

**Comment by duncanista** [comment]
Hey @xonstone,

While manually testing, I realized that we were packing `integration_tests` – really bad! Hahaha, this also caused having the dependency in question to be paired as peer. I'm gonna raise a one-liner which removes it from being packaged, since the `integration_tests` somehow rely on this dependency.

**Review by duncanista — APPROVED** [review]
LGTM.

Verified:
- Unit tests passing,
- Integration tests passing,
- Lint passes,
- Manually tested.

Will merge after fixing packaging adding `integration_tests`

**Pull request merged**